hp2ps depends on "ghcautoconf.h" and "ghcplatform.h", but does not yet need it.

```
Reading shake-build/cfg/system.config...
Reading package dependencies...
| Successfully generated 'utils/hp2ps/stage0/package-data.mk'.
Reading utils/hp2ps/stage0/package-data.mk...
Reading dependencies from utils/hp2ps/stage0/build/.dependencies...
/----------
| Running GccM Stage0 with arguments:
|   ... skipping 13 arguments ...
|   utils/hp2ps/Utilities.c
\----------

In file included from utils/hp2ps/Utilities.c:1:
In file included from utils/hp2ps/Main.h:4:
includes/ghcconfig.h:4:10: fatal error: 'ghcautoconf.h' file not found
#include "ghcautoconf.h"
         ^
1 error generated.
shakeArgsWith   0.000s    0%                           
Function shake  0.019s    5%  =                        
Database read   0.000s    0%                           
With database   0.000s    0%                           
Running rules   0.314s   94%  ======================== 
Total           0.334s  100%                           
Error when running Shake build system:
* inplace/bin/hp2ps
* utils/hp2ps/stage0/build/Utilities.o
* OracleQ (DependenciesKey ("utils/hp2ps/stage0/build/.dependencies","utils/hp2ps/stage0/build/Utilities.o"))
* utils/hp2ps/stage0/build/.dependencies
* utils/hp2ps/stage0/build/Utilities.c.deps
user error (Development.Shake.cmd, system command failed
Command: /usr/bin/clang -E -MM -Iincludes -Iincludes/dist -Iincludes/dist-derivedconstants/header -Iincludes/dist-ghcconstants/header -Iutils/hp2ps/stage0/build/autogen -MF utils/hp2ps/stage0/build/Utilities.c.deps -MT utils/hp2ps/stage0/build/Utilities.o -x c utils/hp2ps/Utilities.c
Exit code: 1
Stderr:
In file included from utils/hp2ps/Utilities.c:1:
In file included from utils/hp2ps/Main.h:4:
includes/ghcconfig.h:4:10: fatal error: 'ghcautoconf.h' file not found
#include "ghcautoconf.h"
         ^
1 error generated.
)
```

and after `./shake-build/build.sh includes/ghcautoconf.h`

```
Reading shake-build/cfg/system.config...
Reading package dependencies...
Reading utils/hp2ps/stage0/package-data.mk...
Reading dependencies from utils/hp2ps/stage0/build/.dependencies...
/----------
| Running GccM Stage0 with arguments:
|   ... skipping 13 arguments ...
|   utils/hp2ps/Utilities.c
\----------

In file included from utils/hp2ps/Utilities.c:1:
In file included from utils/hp2ps/Main.h:4:
includes/ghcconfig.h:5:10: fatal error: 'ghcplatform.h' file not found
#include "ghcplatform.h"
         ^
1 error generated.
shakeArgsWith   0.000s    0%                           
Function shake  0.018s   12%  ===                      
Database read   0.006s    3%  =                        
With database   0.000s    0%                           
Running rules   0.119s   83%  =========================
Total           0.143s  100%                           
Error when running Shake build system:
* inplace/bin/hp2ps
* utils/hp2ps/stage0/build/Utilities.o
* OracleQ (DependenciesKey ("utils/hp2ps/stage0/build/.dependencies","utils/hp2ps/stage0/build/Utilities.o"))
* utils/hp2ps/stage0/build/.dependencies
* utils/hp2ps/stage0/build/Utilities.c.deps
user error (Development.Shake.cmd, system command failed
Command: /usr/bin/clang -E -MM -Iincludes -Iincludes/dist -Iincludes/dist-derivedconstants/header -Iincludes/dist-ghcconstants/header -Iutils/hp2ps/stage0/build/autogen -MF utils/hp2ps/stage0/build/Utilities.c.deps -MT utils/hp2ps/stage0/build/Utilities.o -x c utils/hp2ps/Utilities.c
Exit code: 1
Stderr:
In file included from utils/hp2ps/Utilities.c:1:
In file included from utils/hp2ps/Main.h:4:
includes/ghcconfig.h:5:10: fatal error: 'ghcplatform.h' file not found
#include "ghcplatform.h"
         ^
1 error generated.
)
```

--- a/README.md
+++ b/README.md
@@ -55,9 +55,6 @@ git submodule update --init
 git clone git://github.com/snowleopard/shaking-up-ghc shake-build
 ./boot
 ./configure --with-gcc=$(which clang) # See #26
-./shake-build/build.sh includes/ghcautoconf.h # See #48
-./shake-build/build.sh includes/ghcplatform.h # See #48
-cp utils/hsc2hs/template-hsc.h inplace/lib/template-hsc.h # See #44
 ./shake-build/build.sh
 ```
 

--- a/src/Rules/Dependencies.hs
+++ b/src/Rules/Dependencies.hs
@@ -19,12 +19,14 @@ buildPackageDependencies _ target @ (PartialTarget stage pkg) =
         (buildPath <//> "*.c.deps") %> \out -> do
             let srcFile = dropBuild . dropExtension $ out
             when (pkg == compiler) . need $ platformH : includesDependencies
+            when (pkg == hp2ps) . need $ ["includes/ghcautoconf.h", "includes/ghcplatform.h"]
             need [srcFile]
             build $ fullTarget target (GccM stage) [srcFile] [out]
 
         hDepFile %> \out -> do
             srcs <- interpretPartial target getPackageSources
             when (pkg == compiler) . need $ platformH : includesDependencies
+            when (pkg == hp2ps) . need $ ["includes/ghcautoconf.h", "includes/ghcplatform.h"]
             -- TODO: very ugly and fragile; use gcc -MM instead?
             let extraDeps = if pkg /= compiler then [] else fmap (buildPath -/-)
                    [ "primop-vector-uniques.hs-incl"

